### PR TITLE
feat(alerts): Replace aggregation with aggregate

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This does two things

 * Removes the `aggregation` field and adds a new `aggregate` field, which is configured via the metricField component.

 * Introduces a `dataset` field which can select between errors and transactions. This changes what aggregates can be configured. This is currently behind an internal feature flag

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx [hunk 1]
```diff
@@ -61,10 +56,7 @@ export default class DetailsBody extends React.Component<Props> {
       id: undefined,
       name: (incident && incident.title) || '',
       fields: ['issue', 'count(id)', 'count_unique(user.id)'],
-      orderby:
-        incident.alertRule.aggregation === AlertRuleAggregations.UNIQUE_USERS
-          ? '-count_unique_user_id'
-          : '-count_id',
+      orderby: `-${incident.alertRule.aggregate}`,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/alerts/details/body.tsx:59]: Need confirmation that this is right and that i can do this. @wedamija. I feel it is.. not right

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx:59]: It's right.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx:59]: Are you sure? this will be `-percentile(transaction.duration, 0.5)` that's right?

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx:59]: Ah.

you'll need to run it through https://github.com/getsentry/sentry/blob/cc3a93b7675bca623475f2bb0ba8b56b6e4f6228/src/sentry/static/sentry/app/utils/discover/fields.tsx#L373-L383

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx [hunk 2]
```diff
@@ -101,51 +104,64 @@ class RuleConditionsForm extends React.PureComponent<Props, State> {
       <Panel>
         <PanelHeader>{t('Configure Rule Conditions')}</PanelHeader>
         <PanelBody>
+          <Feature
+            requireAll
+            features={[
+              'organizations:transaction-events',
+              'organizations:internal-catchall',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx:111]: Are you using this to hide these controls from alpha users while you finish off the feature?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx:111]: yup

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**